MentorMatchmaker/MentorMatchmaker/MatchMakingDomainOperations.fs

```diff
@@ -222,38 +238,309 @@ module Matchmaking =
+        | Ok matches ->
+            let fileContents =
+                [
+                    [ "Mentor"; "Mentee"; "Topic"; "Mentor could handle more" ]
+                    for pair in matches do
+                        [
+                            pair.MatchedMentor.MentorInformation.Fullname + $" ({pair.MatchedMentor.MentorInformation.EmailAddress}) ({pair.MatchedMentor.MentorInformation.SlackName})"
+                            pair.MatchedMentee.MenteeInformation.Fullname + $" ({pair.MatchedMentee.MenteeInformation.EmailAddress}) ({pair.MatchedMentee.MenteeInformation.SlackName})"
+                            pair.FsharpTopic.Category.CategoryName
+                            pair.CouldMentorHandleMoreWork.ToString()
+                            dumpMeetingTimes pair.MeetingTimes
+                        ]
+                ]
+                |> List.map (String.concat ";")
+                |> String.concat "\n"
+
+
+
+            let outputFileNameMatchesCsv = FileManager.saveVersionedCsv "matches" fileContents
+
+            Ok outputFileNameMatchesCsv
+
+        | Error err ->
+            printfn $"Error: {err}"
+            Error err
+
+
+    let unmatchedDataDump applicantsJsonPath =
+        match FileManager.readJson<Applicants> applicantsJsonPath with
+        | Ok applicants ->
+            let fileContents = applicants |> generateUnmatchedApplicantsDumpFileContents
+
+            let outputFileName = FileManager.saveVersionedText "applicationDataDumpUnmatchedApplications" fileContents
+
+            Ok outputFileName
+
+        | Error err ->
+            printfn $"Error: {err}"
+            Error err
+
+    let unmatchedPermutationsDataDump applicantsJsonPath =
+        match FileManager.readJson<Applicants> applicantsJsonPath with
+        | Ok applicants ->
+            let fileContents = applicants |> generateUnmatchedApplicantPairingPermutationsFileContents
+
+            let outputFileName = FileManager.saveVersionedText "applicationDataDumpUnmatchedApplicationsPairingPermutations" fileContents
+            
+            Ok outputFileName
+
+        | Error err ->
+            printfn $"Error: {err}"
+            Error err
+
+    let generateExampleEmails matchesJsonPath = 
+        match FileManager.readJson<ConfirmedMentorshipApplication list> matchesJsonPath with
+        | Ok matches ->
+            let fileContents = matches |> EmailGenerationService.generateEmailExamplesForMatches
+
+            let outputFileName = FileManager.saveVersionedText "exampleEmailsDump" fileContents
+
+            Ok outputFileName
+
+        | Error err ->
+            printfn $"Error: {err}"
+            Error err
+
+    
+    let sendEmailsMatched matchesJsonPath =
+        match FileManager.readJson<ConfirmedMentorshipApplication list> matchesJsonPath with
+        | Ok matches ->
+            // Do something
+            for pair in matches do
+                use client = createSmtpClient ()
```

Comment:
I don't think you need to create an SMTP connection every time you find a match; could be moved outside the loop.

Reply:
Done.